include/aws/auth/signing.h

```diff
@@ -98,11 +56,15 @@ int aws_signer_sign_request(
  * In all cases, the signing result will tell exactly what header and/or query params to add to the request
  * to become a fully-signed AWS http request.
  *
- * These restrictions can be relaxed, if necessary, in the future, but they don't unreasonable and
- * relaxing them adds non-trivial complexity.
+ * These restrictions can be relaxed, if necessary, in the future.
  */
 AWS_AUTH_API
-struct aws_signer *aws_signer_new_aws(struct aws_allocator *allocator);
+int aws_sign_request_aws(
```

Comment:
Worth having a typedef to cover all signers?
`typedef int aws_signing_fn(.....);`